tools/ci_build/build.py

```diff
@@ -1312,140 +1312,6 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
-        cmd_base = []
-        if provider:
-            cmd_base += ["-e", provider]
-
-        if num_parallel_tests != 0:
-            cmd_base += ['-c', str(num_parallel_tests)]
-
-        if num_parallel_models > 0:
-            cmd_base += ["-j", str(num_parallel_models)]
-
-        # onnx test
-        if os.path.exists(onnx_test_data_dir):
-            onnx_test_cmd = cmd_base + [onnx_test_data_dir]
-            run_subprocess([exe] + onnx_test_cmd, cwd=cwd, dll_path=dll_path)
-
-        # model test
-        # TensorRT can run most of the model tests, but only part of
-        # them is enabled here to save CI build time.
-        if config != 'Debug' and os.path.exists(model_dir):
-            model_dir_opset8 = os.path.join(model_dir, "opset8")
-            model_dir_opset8 = glob.glob(os.path.join(
-                model_dir_opset8, "test_*"))
-            model_dir_opset10 = os.path.join(model_dir, "opset10")
-            model_dir_opset10 = glob.glob(
-                os.path.join(model_dir_opset10, "tf_inception_v1"))
-            for dir_path in itertools.chain(model_dir_opset8,
-                                            model_dir_opset10):
-                model_test_cmd = cmd_base + [dir_path]
-                run_subprocess(
-                    [exe] + model_test_cmd, cwd=cwd, dll_path=dll_path)
-
-
-def openvino_run_onnx_tests(build_dir, configs, onnx_test_data_dir,
-                            provider, num_parallel_models,
-                            num_parallel_tests=0):
-    """openvino function to run onnx tests and model tests
-    """
-    for config in configs:
-        cwd = get_config_build_dir(build_dir, config)
-        if is_windows():
-            exe = os.path.join(cwd, config, 'onnx_test_runner')
-            model_dir = os.path.join(cwd, "models")
-        else:
-            exe = os.path.join(cwd, 'onnx_test_runner')
-            model_dir = os.path.join(build_dir, "models")
-
-        cmd = ['-o', '0']
-        if provider:
-            cmd += ["-e", provider]
-
-        if num_parallel_tests != 0:
-            cmd += ['-c', str(num_parallel_tests)]
-
-        if num_parallel_models > 0:
-            cmd += ["-j", str(num_parallel_models)]
-
-        # onnx test
-        if os.path.exists(onnx_test_data_dir):
-            cmd.append(onnx_test_data_dir)
-            run_subprocess([exe] + cmd, cwd=cwd)
-
-        # model test
-        # OpenVINO can run most of the model tests, but only part of
-        # them are enabled here to save CI build time.
-        if config != 'Debug' and os.path.exists(model_dir):
-            model_dir_opset8 = os.path.join(model_dir, "opset8")
-            model_dir_opset8 = glob.glob(os.path.join(
-                model_dir_opset8, "test_*"))
-            model_dir_opset10 = os.path.join(model_dir, "opset10")
-            model_dir_opset10 = glob.glob(os.path.join(
-                model_dir_opset10, "*v1*"))
-            for dir_path in itertools.chain(model_dir_opset8,
```

Comment:
how do we retain the ability to run only models in certain directories with the new scheme?
i glanced at the other PR and it looks like we disable models by name there. but it doesn't seem to include directory (like opset8, opset10) so does that mean if a model exists in both opset8 and 10, and we want to disable only opset8, we can't?

Reply:
It still can, and it is even better. It filters the models based on the content of the model file, instead of path name. 

See: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/providers/cpu/model_tests.cc#L56